src/compiler/checker.ts

```diff
@@ -9349,6 +9349,10 @@ namespace ts {
             if (!links.type) {
                 const targetSymbol = resolveAlias(symbol);
                 const exportSymbol = symbol.declarations && getTargetOfAliasDeclaration(getDeclarationOfAliasSymbol(symbol)!, /*dontResolveAlias*/ true);
+                const declaredType = firstDefined(exportSymbol?.declarations, d => isExportAssignment(d) ? tryGetTypeFromEffectiveTypeNode(d) : undefined);

```

Comment:
please inline this into the conditional expression, below the isDuplicatedCommonJSExport/autoType case.

Reply:
@sandersn Thanks for the feedback.

> I meant that I wanted to see an error for assigning { a } ==> { a, b } but not for { a, b, c } ==> { a, b }. Sorry about not being clearer.

Similar to variable assignments, like so [Playground](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgGIHt3IN4ChkHJwBcyIArgLYBG0A3LgL664A2EYRAjKRlgLw4ipLgBpk1EckZ1kAejnIAolCjooLdp2rJB2YcjESpMthyIAmXpl0TZC5AHkA1sgD8H3EA)?

Reply:
More like [this Playground](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgGIHt3IN4ChkHJwBcyIArgLYBG0A3MtaRTdLgL664A2EYRARlIZ0PPkWQBeHEVIDknXvzjUEUmSWQCANIzm6EchXVxx50uAwD0V5AFEoUdFFPmiq67YDyAa2QB+QNwgA)

Reply:
Does [`checkJsdocTypeTagOnExportAssignment6`](https://github.com/microsoft/TypeScript/blob/db74fad86556e37298ae3def41fbddec8e6f14ea/tests/baselines/reference/checkJsdocTypeTagOnExportAssignment6.errors.txt) cover that case partially?

Reply:
I've updated tests [`checkJsdocTypeTagOnExportAssignment6`](https://github.com/microsoft/TypeScript/blob/0df00efba7fd2b55d227acbe544cc90ec056a9ad/tests/cases/compiler/checkJsdocTypeTagOnExportAssignment6.ts), [`checkJsdocTypeTagOnExportAssignment7`](https://github.com/microsoft/TypeScript/blob/eaa42398d2dd065741d9e0b5254099ac6172065f/tests/cases/compiler/checkJsdocTypeTagOnExportAssignment7.ts).